File: 121_best_time_to_buy_and_sell_stock/maxProfit.cpp
#include <vector>

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0;
        int current_min = prices[0];

        for (int price : prices) {
            int profit = price - current_min;
            current_min = min(current_min, price);
            max_profit = max(max_profit, profit);
        }

        return max_profit;
    }
};

File: 121_best_time_to_buy_and_sell_stock/max_profit.rs
use std::cmp;

impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut max_profit = 0;
        let mut min_price = prices[0];

        for i in 1..prices.len() {
            let price = prices[i];
            let profit = price - min_price;
            max_profit = cmp::max(max_profit, profit);
            min_price = cmp::min(min_price, price);
        }

        return max_profit;
    }
}
